### ISSUE TITLE
Could we make coq-ltac2 a dummy package in 8.11 instead of requiring Coq < 8.11?

### ISSUE THREAD

**Issue by cpitclaudel** [coq/coq#12367]
#### Description of the problem

I'm trying to support 8.9 through 8.11 in the same project.  I use Ltac2, so I have a dependency on the coq-ltac2 package.  But currently that package requires Coq < 8.11.

Could the coq-ltac2 be marked as compatible with 8.11 and made to install a dummy package there?  Or is there a simpler solution that I overlooked?

cc @ppedrot

**Comment by SkySkimmer** [comment]
Can't you depend on coq-ltac2 || (Coq >= 8.11)?

Gaëtan Gilbert

**Comment by JasonGross** [comment]
According to https://opam.ocaml.org/doc/Manual.html, the correct syntax is probably `"coq-ltac2" | "coq" { >= "8.11~" }` (the `~` allows capturing also the beta versions, I believe)

**Comment by cpitclaudel** [comment]
Thanks, I should have looked there.  I only looked at https://dune.readthedocs.io/en/stable/dune-files.html#package, and AFAICT dune doesn't support that.  Am I missing something?

**Comment by JasonGross** [comment]
Indeed, it looks like this feature might be missing from dune.  @ejgallego?

**Comment by ejgallego** [comment]
I don't recall the status of that, at least on master you should be able to do:
```
(package
 (name cohttp)
 (synopsis "An OCaml library for HTTP clients and servers")
 (description "A longer description")
 (depends
  (alcotest :with-test)
  (dune (> 1.5))
  (foo (or :dev (> 1.5) (< 2.0)))
  (uri (>= 1.9.0))
  (uri (< 2.0.0))
  (fieldslib (> v0.12))
  (fieldslib (< v0.13))))

```

**Comment by cpitclaudel** [comment]
This is an OR between versions, but we need an OR between packages.

**Comment by ejgallego** [comment]
> This is an OR between versions, but we need an OR between packages.

If that's not yet implemented I'd suggest you file a feature request and hopefully will solved soon. I can check what's the status of this but not today sorry.

**Comment by cpitclaudel** [comment]
https://github.com/ocaml/dune/issues/3497

**Comment by ejgallego** [comment]
Thanks a lot for the report @cpitclaudel , regarding the opam issue, I suggest to re-submit at https://github.com/coq/opam-coq-archive/issues as it is an issue of the repos.

**Comment by Zimmi48** [comment]
Isn't it the whole issue? It is easy to transfer issues between repos of the same organization @ejgallego.
